app/controllers/alaveteli_pro/stripe_webhooks_controller.rb

```diff
@@ -67,4 +68,33 @@ def notify_exception(error)
     end
   end
 
+  # ignore any that don't match our plan namespace
+  def filter_hooks
+    plans = []
+    case @stripe_event.data.object.object
+    when 'subscription'
+      plans = get_plan_names(@stripe_event.data.object.items)
+    when 'invoice'
+      plans = get_plan_names(@stripe_event.data.object.lines)
+    end
+
+    # ignore any plans that don't start with our namespace
+    plans.delete_if { |plan| !plan_matches_namespace(plan) }
+
+    if plans.empty?
+      # reject it and throw it away, the emails are just noise
+      render json: { error: 'Does not appear to be one of our plans' },
+             status: 401
+      return
+    end
+  end
+
+  def plan_matches_namespace(plan_name)
+    (AlaveteliConfiguration.stripe_namespace == '' ||
+     plan_name =~ /^#{AlaveteliConfiguration.stripe_namespace}/)
+  end
+
+  def get_plan_names(items)
+    items.map { |item| item.plan.name if item.plan }.compact.uniq
```

Comment:
The namespace isn't necessarily part the plan name which is something more descriptive. Instead you want to be pulling out the plan ID here and probably need to look into why this wasn't caught by the specs